src/inference/rejection.js

```diff
@@ -33,28 +41,49 @@ module.exports = function(env) {
     this.oldCoroutine = env.coroutine;
     env.coroutine = this;
 
-    if (!_.isNumber(this.numSamples) || this.numSamples <= 0) {
-      throw new Error('samples should be a positive integer.');
-    }
-
     if (this.incremental && this.maxScore > 0) {
       util.warn('Rejection: Reduce maxScore to zero for better performance.');
     }
   }
 
   Rejection.prototype.run = function() {
+    if (!_.isNumber(this.numSamples) || this.numSamples <= 0) {
+      return this.error('"samples" should be a positive integer.');
+    }
+    var elapseSec = (Date.now() - this.startTime) / 1000.0;
+    if (elapseSec > 2) {
```

Comment:
I see. I'm not sure what's best here. One idea that doesn't require an extra flag would be to replace the conditional with `(this.minSampleRate > 0) && (elapseSec > 2)` and to set `this.minSampleRate` to 0 once the `sampleRate` check succeeds. This would make it explicit that you're only checking once after about two seconds and then accept any sample rate.